test/test_torch.py

```diff
@@ -939,6 +944,385 @@ def test_arange(self):
+        smaller_ndims = random.randint(1, ndims)
+        dims_small = []
+        dims_large = []
+        for i in range(ndims - 1, -1, -1):
+            j = random.randint(1, 3)
+            if j == 1:  # no reduced singleton dimension
+                ds = dims_full[i]
+                dl = dims_full[i]
+            elif j == 2:  # larger may have reduced singleton dimension
+                ds = dims_full[i]
+                dl = 1 if len(dims_small) < smaller_ndims else dims_full[i]
+            elif j == 3:  # smaller may have reduced singleton dimension
+                ds = 1
+                dl = dims_full[i]
+            dims_large = [dl] + dims_large
+            if len(dims_small) < smaller_ndims:
+                dims_small = [ds] + dims_small
+        return (dims_small, dims_large, dims_full)
+
+    @staticmethod
+    def _test_broadcast(self, cast):
+
+        # all functions
+        fns = [
+            "dist", "atan2", "pow", "lerp", "add",
+            "sub", "mul", "div", "fmod", "remainder",
+            "eq", "ge", "gt", "le", "lt", "max", "min", "ne",
+            "addcdiv", "addcmul", "masked_copy", "masked_fill",
+            "map", "map2", "copy"
+        ]
+        # functions with no torch. equivalent
+        fns_no_torch = ["sub", "masked_copy", "masked_fill", "map", "map2", "copy"]
+        # functions with no inplace equivalent
+        fns_no_inplace = ["dist", "max", "min"]
+        # functions with no inplace cuda implementation
+        fns_no_inplace_cuda = ["map", "map2"]
+        # functions with no out-of-place tensor version
+        fns_no_out_place = ["masked_copy", "masked_fill", "map", "map2", "copy"]
+        # functions with fallback to equal nElem behavior
+        fns_fallback = ["add", "sub", "div", "mul", "pow", "fmod", "remainder",
+                        "eq", "ge", "gt", "le", "lt", "max", "min", "ne",
+                        "addcdiv", "addcmul", "masked_copy", "masked_fill",
+                        "map", "map2", "copy"]
+        # functions with three tensor arguments
+        fns_3_args = ["addcdiv", "addcmul", "map2"]
+
+        for fn in fns:
+            (dims_small, dims_large, dims_full) = TestTorch._select_broadcastable_dims(self)
+            small = torch.randn(*dims_small).float()
+            small = cast(small)
+            large = torch.randn(*dims_large).float()
+            large = cast(large)
+            smallExpanded = small.expand(*dims_full)
+            largeExpanded = large.expand(*dims_full)
+            small2 = None
+            small2Expanded = None
+            if fn in fns_3_args:
+                # create another smaller tensor
+                (dims_small2, _, _) = TestTorch._select_broadcastable_dims(self, dims_full)
+                small2 = torch.randn(*dims_small2).float()
+                small2 = cast(small2)
+                small2Expanded = small2.expand(*dims_full)
+
+            if fn not in fns_no_out_place:
```

Comment:
yes, good suggestion.   I was going to see if I could get rid of the exceptions (e.g. just add a max_), but the issue become large enough by itself.